tests/rust/test_rust_backend.py

```
@@ -0,0 +1,674 @@
+    def test_mocked_streaming_output(self, mock_popen, rust_backend, sample_rust_code, temp_dir):
+        """Test streaming output handling with mocked Popen."""
+        mock_process = MagicMock()
+        mock_process.stdout.readline.side_effect = [b"Line 1\n", b"Line 2\n", b""]
+        mock_process.stderr.readline.side_effect = [b""]
+        mock_process.poll.return_value = 0
+        mock_process.wait.return_value = 0
+        mock_popen.return_value = mock_process
+        source_file = temp_dir / "main.rs"
+        source_file.write_text(sample_rust_code)
+        if hasattr(rust_backend, 'compile_with_streaming'):
+            result = rust_backend.compile_with_streaming(str(source_file))
```

Ruff (0.11.9)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)